File: dynamic_listings/python/default/containers/particle_emitter.py
class Emitter:
    """
    This is a simple particle emitter, it contains a list
    of particles and it updates and manages them
    """
    origin: Vector2 = None
    particles: list[Particle] = []

    def __init__(self, location: Vector2) -> None:
        self.origin = location
        self.particles = [Particle() for _ in range(8)]  # We prepare 8 particles

    def update(self, dt: float) -> None:
        # Update the entire system, by updating each particle
        for particle in self.particles:
            if not particle.is_dead():
                particle.update(dt)

File: dynamic_listings/lua/default/containers/particle_emitter.lua
-- This is a simple particle emitter, it contains a list
-- of particles and it updates and manages them
Emitter = {}

function Emitter:new(o)
    o = o or {
        origin = {x=0, y=0},
        particles = {}
    }
    setmetatable(o, self)
    self.__index = self
    return o
end

function Emitter:update(dt)
    -- Update the entire system, by updating each particle
    for i = 1, #self.particles do
        local particle = self.particles[i]
        if (not particle.is_dead()) then
            particle.update(dt);
        end
    end
end
